### PR TITLE
refactor: simplify over-engineered code paths

### PR DESCRIPTION
**Pull request by AllenYolk**
## Summary

- Apply a repository-wide Ponytail audit and simplify confirmed over-engineering while preserving public contracts and useful example workflows.
- Consolidate duplicate reset, traversal, operation-counter, device, model-invocation, and dataset-conversion rules at their existing owners.
- Delete dead wrappers, mirrored state, speculative compatibility options, obsolete commented implementations, and the unused direct `einops` dependency.
- Tighten invalid-state handling and fix issues exposed by the audit, including OTTT drop-rate forwarding and pipeline partition over-splitting.
- Update public API changelog and distributed-training documentation for the intentional removals.

The final PR diff changes 110 files with 585 insertions and 1,022 deletions, a net reduction of 437 lines. Under `spikingjelly/`, it changes 90 files with 431 insertions and 982 deletions, a net reduction of 551 lines.

## Intentional API changes

- Remove the ineffective `SpikeZIPQANNRecipe.strict` option.
- Remove the unused `model_family` argument from distributed `analyze()`.
- Remove the unused `tau_trace` argument from `mstdpet_linear_single_step()`.

## Validation

- Remote g2 targeted suite before maintainer review: 465 passed.
- Remote g2 full suite before maintainer review: 2,140 passed, 61 skipped, 39 warnings in 601.37s.
- Remote g2 maintainer-review targeted tests: 9 passed; final post-docstring check: 8 passed.
- Remote g2 maintainer-review affected-module regression: 443 passed, 27 skipped.
- Former flaky pipeline regression: passed five consecutive runs.
- Remote g2 clean Sphinx HTML build: succeeded; 1,450 pre-existing warnings.
- `uv lock --check`.
- Generated changelog consistency check.
- Changed-file Ruff checks: F401, F811, F821, F841, E722.
- `git diff --check`.

The branch is rebased onto the current `origin/master`; maintainer feedback has been applied without adding compatibility layers or new dependencies.

## Summary by CodeRabbit

* **New Features**
  * Added automatic device selection for ANN-to-SNN conversion.
  * Added support for pipeline-parallel distributed plans.
  * Added public utilities for module replacement, model invocation, and CUDA device normalization.
  * Added public export support for the OTTT Spiking VGG model.

* **Bug Fixes**
  * Fixed local-rank validation and pipeline state-reset behavior.
  * Corrected dropout-rate handling for OTTT Spiking VGG.
  * Improved case-insensitive rate-coding mode validation.

* **API Updates**
  * Simplified analysis, conversion, and learning interfaces by removing unused arguments.

### CODE REVIEW

File: spikingjelly/activation_based/ann2snn/examples/cnn_mnist.py [hunk 1]
```diff
@@ -310,25 +308,6 @@ def main(args):
...
-    #     model.train()
-    #     for img, label in train_data_loader:
-    #         optimizer.zero_grad()
-    #         out = model(img.to(device))
-    #         loss = loss_function(out, label.to(device))
-    #         loss.backward()
-    #         optimizer.step()
-    #     torch.save(model.state_dict(), "SJ-mnist-cnn_model-sample.pth")
-    #     print("Epoch: %d" % epoch)
-    #     print("Validating Accuracy: %.3f" % val(model, device, train_data_loader))
```
Comment (AllenYolk) [inline spikingjelly/activation_based/ann2snn/examples/cnn_mnist.py:330]: These comments should not be removed. Users can uncomment these code to train the ANN from scratch.

File: spikingjelly/activation_based/ann2snn/recipes/base.py [hunk 2]
```diff
@@ -19,6 +19,14 @@
 ]
 
 
+def _replace_submodule(
```
Comment (AllenYolk) [inline spikingjelly/activation_based/ann2snn/recipes/base.py]: This helper function is not used locally in this file. Instead, it is used elsewhere. It should not be a private function.

File: spikingjelly/activation_based/distributed/pipeline/runtime.py [hunk 3]
```diff
@@ -242,12 +235,11 @@ def _measure_module_cost(module: nn.Module, input_value: Any) -> Tuple[Any, floa
             start_time = time.perf_counter()
             output_value = module(input_value)
             elapsed_ms = (time.perf_counter() - start_time) * 1000.0
-        _reset_collected_modules(reset_modules)
```
Comment (AllenYolk) [inline spikingjelly/activation_based/distributed/pipeline/runtime.py:245]: Is it safe to remove the reset operation here?

File: spikingjelly/activation_based/op_counter/base.py [hunk 4]
```diff
@@ -46,6 +46,14 @@ def _collect_tensors(tree: Any) -> list[torch.Tensor]:
     return [x for x in flat if torch.is_tensor(x)]
 
 
+def _call_model(model: nn.Module, inputs):
```
Comment (AllenYolk) [inline spikingjelly/activation_based/op_counter/base.py]: The function is never use in this file. Instead, it is used elsewhere. So it should not be a private function.

File: spikingjelly/activation_based/triton_kernel/triton_utils.py [hunk 5]
```diff
@@ -351,6 +351,13 @@ def torch_dtype_for_triton_neuron_compute_dtype_id(dtype_id: int) -> torch.dtype
     return triton_neuron_dtype_id_to_torch_dtype(dtype_id)
 
 
+def _normalize_cuda_device(device) -> torch.device:
```
Comment (AllenYolk) [inline spikingjelly/activation_based/triton_kernel/triton_utils.py]: Used elsewhere. So it should not be a private function.

File: PONYTAIL_AUDIT.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
+# Ponytail audit
```
Comment (AllenYolk) [inline PONYTAIL_AUDIT.md]: This file should not be committed.
